src/transformers/generation/utils.py

```diff
@@ -3222,6 +3223,16 @@ def _sample(
         unfinished_sequences = torch.ones(batch_size, dtype=torch.long, device=input_ids.device)
         model_kwargs = self._get_initial_cache_position(input_ids, model_kwargs)
 
+        def model_forward(model, *args, **kwargs):
+            return model.forward(*args, **kwargs)
```

Comment:
@ArthurZucker This PR breaks some tests on PEFT :(

I checked why that is exactly, and I found that it has nothing to do with the compilation. The sole reason is that on 2nd iteration, we use this function, which effectively calls:

`self.forward(**model_inputs, return_dict=True)`

whereas on the first iteration (and before the PR), we would call:

`self(**model_inputs, return_dict=True)`

Is there any specific reason why `forward` is used? Using `__call__` looks correct to me.

Reply:
Should be solved with https://github.com/huggingface/transformers/pull/34923 🤗